### ISSUE TITLE
Core/plugin update: 2026-09-16

### ISSUE THREAD

**Issue by curtismchale** [proudcity/wp-proudcity#2935]
## Builds

- [staging](https://console.cloud.google.com/cloud-build/builds;region=global/7f5ac958-4fa5-4e75-bb1e-dff9a27e9b86?project=proudcity-1184)
- [prod](https://console.cloud.google.com/cloud-build/builds;region=global/f9e1de75-1632-4a55-8680-c1c3e1280132?project=proudcity-1184)

## Manual downloads / tag releases

- [x] Gravity Forms 3.1.1.2

## After the image ships

Post-release steps for this round. Neither is optional — both are the kind that fail quietly.

- [ ] **Finish the Gravity Forms DB upgrade.** GF moved 3.1.0.3 -> 3.1.1.2. It ships inside the image via composer, so `upgrader_process_complete` never fires, the DB upgrade falls to a background path, and that path is blocked by a stale `gf_upgrade_lock` — `maybe_clear_previous_upgrade()` only clears a lock whose `to_version` differs from the installed version, so once it reaches that state the site never retries and `gf_db_version` sits behind forever. Stranded 29 prod sites on 2.10.5 (#2892) and 38 on 3.0.2.6 (#2895). The lock clear is the half that breaks the cycle — a forced upgrade alone re-creates the lock in the same wedged state (#2910)
  ```bash
  pck gf-upgrade test all
  pck gf-upgrade prod all
  ```
  Read-only until it prints a plan and you confirm. Confirm it comes back `Nothing to do -- every Gravity Forms site in prod is upgraded and unlocked.`
- [ ] **Run the migration fleet audit** — `bash ~/Workspace/proudcity/proudcity-kubernetes/bin/check-migrations.sh`. Yoast 28.4 added a schema migration this round, so there is something real to compare against. Detail in the Migration check section below

## Migration check

Runs on **every** update, not just the plugins listed under Notes. A plugin with no migrations today can add one in any release, and a migration that silently fails on a subset of sites is invisible without checking. That is exactly what happened with WP-Stateless (#2889): migrations shipped, ran on 111 sites, silently did not finish on 41 others, and nobody noticed for about two years.

- [x] `git diff composer.json` on the update branch to list which plugins actually moved and from/to what. 62 of 64 are exact-pinned and `composer.lock` is gitignored, so `composer.json` **is** the version record
- [ ] For each plugin that moved, check whether the new version ships a data migration or a schema-version bump. Look for a `migrations/` directory, a `*_db_version` / `*_DB_VERSION` constant, or use of `WP_Background_Process`. If the plugin is in `~/Sites/proudtest/wp-content/plugins/` after `composer update`, diff it there
- [ ] If anything moved, note it here so the post-release check below has something to compare against

After the release has rolled out:

- [ ] Run the fleet audit and confirm no new drift:
  ```bash
  bash ~/Workspace/proudcity/proudcity-kubernetes/bin/check-migrations.sh
  ```
  Exit `0` clean, `1` drift found. It compares every site's migration and schema-version options against the fleet consensus, so it catches migrations that failed on a subset without needing to know which plugins have migrations. See proudcity-kubernetes#migration-watcher.
- [ ] Anything it flags: **investigate before running anything.** Of the first five options triaged this way (#2892), exactly one was a migration that needed running — the rest were orphaned options from uninstalled plugins, an inactive plugin, and a plugin version mismatch. Read the owning plugin's upgrade code first
- [ ] Known-benign outliers go in `proudcity-kubernetes/bin/check-migrations.exclude` **with the reason**

The `migration-watcher` CronJob runs this weekly and files a `migration-drift` issue on its own, so this step is a faster feedback loop rather than the only safety net.

### Findings for this round (2026-09-16)

`git diff composer.json` on `issue/2935` — eight plugins moved, WordPress core did not:

| Plugin | From | To |
| --- | --- | --- |
| disable-comments | 2.8.0 | 2.9.0 |
| elasticpress | 5.3.4 | 5.3.5 |
| proudcity/gravityforms | 3.1.0.3 | 3.1.1.2 |
| google-site-kit | 1.186.0 | 1.187.0 |
| wp-stateless | 4.4.1 | 4.4.3 |
| wp-job-manager | 2.4.6 | 2.4.7 |
| wordpress-seo | 28.3 | 28.5 |

WordPress core is already current — `johnpbloch/wordpress` is pinned to `7.1.0` and the version-check API is still offering 7.1.

`pc kube plugin-outdated` is clean in both namespaces — "Every pinned plugin is on the latest published tag" (prod: 78 unpinned entries tracking HEAD, test: 27). Nothing to pin or roll this round.

Per-plugin migration check:

- [x] **Yoast SEO — new migration, introduced in 28.4.** `src/config/migrations/20260709144332_AddSeoTitleAndMetaDescriptionScores.php` adds two nullable `integer(3)` columns, `seo_title_score` and `meta_description_score`, to the `wp_yoast_indexable` table. Additive and nullable so it is safe in itself, but it is the shape that completes on some sites and silently not on others. **This is the thing to compare against after rollout** — confirm `yoast_migrations_free` moved consistently across the fleet
- [x] **WP-Stateless — clean.** Diffed 4.4.1 against 4.4.3: `static/migrations/` unchanged (still only `20240219175240` and `20240423174109`), `DB::DB_VERSION` in `lib/classes/class-db.php` still `1.2`. No migration pass, no off-hours release
- [x] **Google Site Kit — clean.** The full set of `const DB_VERSION` values is identical between 1.186.0 and 1.187.0
- [x] **disable-comments, ElasticPress, WP Job Manager — clean.** No `migrations/` directory and no `*_db_version` constant in either version

### ElasticPress

- **`wp elasticpress status` and `wp elasticpress stats` lie without `--url`.** Run from WP-CLI with no URL they report `Warning: is not currently indexed.` and `get_index_name()` returns `false`, on a site whose index is fine. That reads exactly like a broken index after a version bump and it is not. Always pass `--url=https://<site>/` when checking EP after an update
- the check that actually answers the question is whether a search query round-trips to Elasticsearch:
  ```bash
  wp eval '$q = new WP_Query(["s"=>"permit","post_type"=>"any","posts_per_page"=>5]);
    echo "found=".$q->found_posts." es=".var_export($q->elasticsearch_success,true)."\n";' \
    --allow-root --path=/app/wordpress --url=https://<site>/
  ```
  `es=true` means EP served it. `es=false` or unset means WP fell back to core search, which is the real failure mode to look for
- cross-check the index itself from inside the pod: `curl -s "http://elasticsearch-client.elasticsearch:9200/_cat/indices?h=index,docs.count,status" | grep <site>`
- `ep_last_sync=false` is normal on test sites and does not mean the index is missing — several test-namespace sites have never run a sync from the admin while still having a populated index

### WP-Stateless

- pinned to an exact version in `composer.json` (`wpackagist-plugin/wp-stateless`, currently 4.4.1), so a bump is always a deliberate edit — run these checks every time that pin moves
- the plugin ships data migrations as files in `static/migrations/`. A new file there means every site needs a migration pass after the release, or sites show the "your data still needs to be updated using this new method" nag and keep reading the legacy storage format. Known migrations as of 4.4.1: `20240219175240` (Update data for Google Cloud files), `20240423174109` (Optimize Compatibility Files) — see #2889
- [x] diff `static/migrations/` between the old and new version. Any new file means a platform-wide migration pass is required
- [x] check whether `DB::DB_VERSION` in `lib/classes/class-db.php` changed (4.4.1 is `1.2`). A bump means a schema change on top of the data migration
- [ ] if either changed, schedule the migration pass as its own off-hours release. `wp stateless migrate` lists status, `wp stateless migrate auto --yes` runs everything still pending (a no-op on an already-current site). Wrap it in `timeout` — `_auto_migrate()` polls the migration state option in an unbounded `while(true)` and will hang the deploy forever if the background process stalls
- **do not put this in `bin/entrypoint.sh`.** Batches are not processed in-process: `Migrator::start_migration()` queues the batch and calls `dispatch()`, a non-blocking loopback POST to `admin-ajax.php`. Entrypoint runs before `exec "$@"` starts `apache2-foreground`, so nothing is listening and the migration registers as started and then sits. Entrypoint also runs on every pod restart and scale-up, not just deploys. `sm_batch_process_cron` is not a fallback — `handle_cron_healthcheck()` also only calls `dispatch()`
- migrations are insert-only (no `DELETE`, no `DROP`, legacy `sm_cloud` postmeta left intact) and each attachment is processed in a transaction against unique keys, so a concurrent upload cannot corrupt data. Still run off-hours: `20240423174109` makes a live GCS API call per row, and edits/deletes of existing media mid-run are the untested path
- verification: after the release, `wp stateless migrate` on a site should list every migration as finished and the upgrade nag should be gone from wp-admin. The fleet-wide version of this is the Migration check at the top of this template
- the 2026-08 pass (#2889) is the worked example of all of the above. Two things it turned up that are not obvious from the code: sites behind the `AUTH_REQUIRED` Basic Auth wall 401 their own loopback, which stops the background processor dead until the `loopback-basic-auth` mu-plugin attaches credentials; and a migration recorded as `skipped` is as finished as one recorded as `finished` — it means `should_run()` returned false for that site

### WP-Stateless Gravity Forms Addon

- after #2831 (Gravity Forms 2.10 broke File Upload sync to GCS) we forked the plugin and patched it for the new JSON storage format. `composer.json` is currently pulling from [proudcity/wp-stateless-gravity-forms-addon](https://github.com/proudcity/wp-stateless-gravity-forms-addon) on `dev-latest` instead of wpackagist.
- our patch branch: [fix/gf-2.10-json-storage](https://github.com/proudcity/wp-stateless-gravity-forms-addon/tree/fix/gf-2.10-json-storage)
- upstream PR: [udx/wp-stateless-gravity-forms-addon#16](https://github.com/udx/wp-stateless-gravity-forms-addon/pull/16) (issue [udx#15](https://github.com/udx/wp-stateless-gravity-forms-addon/issues/15))
- [ ] check if upstream has shipped a release > 0.0.3 that includes the GF 2.10 JSON storage fix. If yes, switch the `proudcity/wp-stateless-gravity-forms-addon` require in `composer.json` back to `wpackagist-plugin/wp-stateless-gravity-forms-addon` at that version and remove the fork repository entry.

## Updates

**Disable Comments** - 2.8.0 -> 2.9.0

Large feature release, not a patch. Worth a look on a test site before it reaches prod.

- New: conditional rules — disable or re-enable comments by taxonomy term, page template, or post age, instead of by post type alone
- New: dry-run preview for Delete Comments, with CSV export of what would be removed
- New: blocked-attempt counter for comment, trackback and REST attempts, reported in Site Health
- New: export and import settings as JSON, with a diff preview before anything is written
- New: WooCommerce product reviews are now a separate setting from ordinary comments
- New: theme conflict scanner for themes that still render a comment form after comments are disabled
- New: Abilities API write support (WordPress 6.9+) — AI assistants and MCP clients can change comment settings and delete comments, gated on the same capabilities as the admin screens
- The advanced tools moved to their own Settings tab
- Developer: `disable-comments delete` gains `--dry-run` and `--export=<file>`; new `disable-comments export` and `disable-comments import` WP-CLI commands

https://wordpress.org/plugins/disable-comments/changelog/

---

**ElasticPress** - 5.3.4 -> 5.3.5

Security release. Both issues are config-dependent and both features are ones we run, so this one should not sit.

- Security: an explicit private-status filter in the admin post list could expose other authors' private posts, including custom private statuses, to users without `read_private_posts` (Protected Content feature)
- Security: the Comments REST search could disclose approved comments on password-protected posts to unauthenticated visitors (Comments feature)
- Security: bumped `postcss-selector-parser` 7.1.0 -> 7.1.5 and `fast-uri` 3.1.5 -> 3.1.7
- Fixed: array-to-string conversion warning in Custom Search Results when post terms are not a string

https://wordpress.org/plugins/elasticpress/changelog/

---

**Gravity Forms** - 3.1.0.3 -> 3.1.1.2

Three releases in the range. 3.1.1 is the feature release and carries a security enhancement; 3.1.1.1 and 3.1.1.2 are the follow-up patches.

_3.1.1_ (2026-09-03)

- Added security enhancements
- Added the `gform_state_lifespan` filter to modify the lifespan of the state during validation
- Added an option on the Export Entries page to select the entry status to export
- Updated admin pages to use a scoped `.gform-admin` root wrapper, adding the `.gform-admin` body class only on pages using components, which reduces style conflicts
- Updated the license check to read the `GF_LICENSE_KEY` constant first and the stored key second; the settings page license field is not editable when set by constant
- Updated the MCP integration to use the canonical WordPress MCP Adapter package to avoid conflicts with other MCP-enabled plugins
- Fixed personal data retention timing out on large entry sets by processing deletions in background batches
- Fixed enter-key date selection in the datepicker navigating to the next field or submitting the form in conversational forms
- Fixed the datepicker F1 key and keyboard navigation
- Fixed the datepicker potentially overwriting a configured minimum or maximum date
- Fixed the default country not being selected when `gform_countries` returns an indexed array
- Fixed fields inside repeaters having hidden labels when the repeater is configured with hidden labels
- Fixed the button spinner staying visible when an AJAX submission is aborted
- Fixed a fatal error from unprefixed filters in the private MCP adapter
- Fixed Forms and Feeds list rows appearing squished on mobile when expanded
- Fixed the Gravity Forms color picker overriding GenerateBlocks color picker styles in the block editor
- Fixed a CSS validation error for form grid styles in Gravity Theme
- Fixed a JavaScript error preventing form display with conditional logic based on a Name field last name input
- Fixed `GFFormsModel::get_current_page_url()` returning the wrong URL when processing a submission via new AJAX under a `Referrer-Policy: no-referrer` header
- Fixed a PHP warning when viewing the changelog

_3.1.1.1_

- Fixed the "value was sanitized" logging statement being recorded on form display for some field types when the field value is empty
- Removed an extra slash from the Jetpack Autoloader file path used to include the canonical WordPress MCP Adapter package
- Fixed the entry list star toggle not being reachable by keyboard or screen readers
- Fixed the "Choose File" button overlapping the field border in conversational forms on Safari
- Added the `gform_addon_pages_using_components` filter to modify which add-on pages use Gravity Forms components

_3.1.1.2_

- Fixed the datepicker day initials using the wrong translation function
- Added merge tag modifiers for the Phone and Time fields so phone numbers can be output unformatted and time values as hour, minute, or am/pm
- Updated the license key setting to stay editable when GoDaddy defines an expired starter license through the `GF_LICENSE_KEY` constant

https://docs.gravityforms.com/gravityforms-change-log/

- [ ] **After the image ships, finish the GF DB upgrade** — see the "After the image ships" checklist at the top of this issue.

---

**Site Kit by Google** - 1.186.0 -> 1.187.0

- Removed the "Site Goals" feature flag — the feature is now fully available
- Added Site Health debug data for the `Content_Events` provider
- Added `pagination_click` event tracking and embedded video event tracking
- Added an "Add Features" admin menu item and a Feature Discovery Hub placeholder
- Added Reader Revenue Manager express setup for connecting existing publications
- Added a "Subscribed users" tab to email reporting settings
- Added the Reader Revenue Manager Organization ID and configured CTAs to Site Health debug info
- Added a warning on the Site Goals widget when the event provider plugin is deactivated
- Updated the email report notice UI and the email subscription panel UX
- Improved page performance by preloading less subscriber content
- Fixed overlapping notifications
- Fixed errors not displaying for email reports
- Fixed a UI loading state bug that prevented errors showing on failed requests
- Fixed `useRefocus()` not firing its callback consistently
- Hid the "Tell us more" link in the Site Goals thumbs up/down feedback

https://wordpress.org/plugins/google-site-kit/changelog/

---

**WP-Stateless** - 4.4.1 -> 4.4.3

Two patch releases, both bug fixes and dependency bumps. No new migrations and no `DB_VERSION` change — see the Migration check above.

_4.4.3_ (2026-09-05)

- Fixed a duplicated `.php` extension in the WooCommerce Extra Product Options add-on file path, which made the add-on always show as uninstalled/inactive on the Stateless Addons screen with an invalid activation link
- Fixed PHP warnings in non-media file synchronization: an undefined `$media` variable after a failed object metadata update, and deprecated dynamic `$client` property creation on PHP 8.2+

_4.4.2_ (2026-09-02)

- Security: `udx/lib-wp-bootstrap` updated to 1.3.5, hardening the admin notice dismissal AJAX handler with nonce verification, a capability check, and an option key restriction
- Updated `guzzlehttp/guzzle` 7.10.0 -> 7.15.5
- Updated `phpseclib/phpseclib` 3.0.48 -> 3.0.57
- Updated `google/protobuf` 4.33.2 -> 4.33.6

https://wordpress.org/plugins/wp-stateless/changelog/

---

**WP Job Manager** - 2.4.6 -> 2.4.7

Several of these are access-control fixes rather than cosmetic ones.

- Applied the View Job capability to the promoted-jobs REST endpoints
- Applied the View Job capability to sitemaps
- Restricted frontend logo attachments to the uploads directory
- Fixed a race condition that could allow the job submission limit to be exceeded (#3013)
- Fixed JobPosting structured data (JSON-LD) being HTML-escaped, so job titles and descriptions read correctly for search engines (#3101)
- Fixed the company logo rendering at full size in the Job Listings admin table on WordPress 7.1 (#3099)
- Cleared unusable attachment values before re-rendering the job submission form
- Indicated WordPress 7.1 compatibility (#3097)
- Updated developer dependencies

https://wordpress.org/plugins/wp-job-manager/changelog/

---

**Yoast SEO** - 28.3 -> 28.5

_28.5_ (2026-09-15)

- Fixed the SEO data optimization looping endlessly when an indexing batch was repeatedly returned without being processed
- Added a link with more information about social previews to the social appearance sections of the editor
- Updated the plugin's app URL and logo used when initiating the MyYoast connection

_28.4_ (2026-09-01)

- Added an option to open the Yoast SEO bulk editor from the Bulk actions menu on the Posts and Pages screens
- Fixed the AI generator's "write more content" tip not showing for posts whose content is mostly block markup, because the markup was being counted as content
- Ships the `AddSeoTitleAndMetaDescriptionScores` migration — see the Migration check above

https://wordpress.org/plugins/wordpress-seo/changelog/

---

## Skipped

- **gtranslate** 3.1.1 -> 5.0.1 — held back. Major version jump that replaces the legacy Google Website Translator widget outright, and the plugin is active on a number of prod sites. Split out to #2936 for impact investigation before it moves. `composer.json` stays on 3.1.1
- **auth0** 3.4.2 -> 4.6.2 — skipped as standing policy, known dashboard compatibility issues with the 4.x line. `composer.json` stays on 3.4.2
- **akismet** — installed but never updated through this process, omitted entirely

**Comment by curtismchale** [comment]
Deployed

Composer: https://github.com/proudcity/wp-proudcity/releases/tag/2026.09.16.1207
Builds: https://github.com/proudcity/proud-recipes/releases/tag/2026.09.16.1207
